src/index.js

```diff
@@ -18,16 +19,22 @@ function normalizeNamespace(fn) {
 }
 
 export function getField(state) {
-  return path => path.split(/[.[\]]+/).reduce((prev, key) => prev[key], state);
+  return path => path.split(/[.[\]]+/).reduce((prev, key) => (prev && prev[key] ? prev[key] : null), state);
 }
 
 export function updateField(state, { path, value }) {
   path.split(/[.[\]]+/).reduce((prev, key, index, array) => {
-    if (array.length === index + 1) {
-      // eslint-disable-next-line no-param-reassign
-      prev[key] = value;
+    // console.log(`BEFORE: array: ${array}  -- index:${index}  -- key:${key} --- prev:`, prev);
+    // console.log('BEFORE: prev[key]:', prev[key]);
+
+    if (index < array.length - 1) {
+      Vue.set(prev, key, prev[key] ? prev[key] : {});
```

Comment:
Is it really necessary to set the value of `prev[key]` to `prev[key]` again? I guess this could affect performance in a negative way.

Reply:
Vue.set is the only way to ensure deep reactivity (https://vuejs.org/v2/guide/reactivity.html).

I could save Vue.set call with a direct assignment if the object property is not responsive (check the existence of a '__ob__' attribute) but I think Vue.set does that anyway.